onnxruntime/core/framework/utils.cc

```diff
@@ -72,6 +72,12 @@ bool ProviderIsCpuBased(const std::string& provider_type) {
          provider_type == onnxruntime::utils::kInternalTestingExecutionProvider;
 }
 
+bool ProviderIsCompiledType(const std::string& provider_type) {
```

Comment:
is this more a 'provider wants 4K alignment'? 

e.g. coreml is compiled but afaik doesn't require 4K alignment. 

Would it be more flexible to have a map of EP to desired alignment, and we have a more generic allocator that takes the alignment as an arg (instead of one hardcoded CPUAllocatorAligned4K)?

Reply:
I agree about the map.

Reply:
I will be adding more providers to the list as the details become available. I renamed the function.